wip-mariocao-blockweight.md

```diff
@@ -0,0 +1,282 @@
+ - **INPUT_SIZE** is the size of each input, which is fixed in size
+ - **OUTPUT_SIZE** is the size of each output, fixed in size
+ - **N** is the number of inputs
+ - **M** is the number of outputs
+
+
+#### Rest of transactions
+
+As the rest of transaction will be allocated in the unbounded bucket as there is no need to weight them.
+
+Commit, reveal and tally transactions will be indirectly limited by previously accepted data requests.
+
+Mint transactions are included by the miners are they are limited to 1 per block.
+
+Additionally, by protocol design, commit and reveal transactions cannot be used as mechanism for transferring value as their output addresses must be equal to the node's identity.
+
+
+#### Estimation of transaction lengths
+
+In order to better understand the impact of assigning different weights to different transaction types it is important to analyze the approximate length of each kind of transaction. While some of them have fixed sizes (e.g., Mint transactions) others may depend on e.g., the number inputs/outputs. The table summarizes both the minimum size of each of these transactions and the average size under common number of inputs/outputs.
+
+- estimation of field lengths
+- do not take into consideration serialization overhead
+
+| Transaction Type |  Minimum size  |       Total size  |  Measurements  |
+| ---------------- | -------------- | ----------------- | -------------- |
+| Mint             |       40 bytes |          76 bytes |    ~  40 bytes |
+| Value Transfer   |      133 bytes |  `N*36+M*36+N*97` |    ~ 205 bytes |
+| Data Request     | 133+data bytes |   `36+N*133+data` |    ~ 400 bytes |
+| Commit           |      376 bytes |       `279+N*133` | ~ W\*412 bytes |
+| Reveal           |      149 bytes |      `149+result` | ~ W\*202 bytes |
+| Tally            | 124+data bytes |  `52+W*36+result` |    ~ 105+W\*36 |
+
+where:
+ - *N* refers to the number of inputs,
+ - *M* to the number of outputs, 
+ - *W* to the number of witnesses specified by the data request,
+ - *data* refers to length of the RADON script to be executed by witnesses,
+ - *result* refers to the length of the bytes retrieved by witnesses.
+
+
+### Specification
+
+Based on the proposal, the following parameter definitions are suggested:
+
+- A maximum block weight of 100K
+- A bounded VTT bucket of 20K
+- A bounded Data Request bucket of 80K
+- A *COMMIT_WEIGHT* of 400
+- A *REVEAL_WEIGHT* of 200
+- A *TALLY_WEIGHT* of 100
+- *beta* is equal to 1
+- *alpha* is equal to 1
+- *gamma* is equal to 10
+
+Currently the known sizes for inputs are outputs are:
+
+- INPUT_SIZE is 133 bytes (97+36)
+- OUTPUT_SIZE is 36 bytes
+
+
+#### System limits
+
+Based on the previously defined parameters, the system will be shaped in different ways:
+
+ - block chain growth
+ - value transfer throughput
+ - data request throughput
+ 
+
+**Blockchain growth**. A maximum block weight of 100K will lead to a theoretical yearly block chain growth of:
```

Comment:
Following my previous comment, here this sentence is not complete right. It is not a theoretical maximum, just an approximation.

Reply:
I think the main problem is implicitly assuming that "weight of 100K" is equal to "100KB", we can say something like "given the maximum block weight of 100K, we can expect an average block size in bytes of 100KB because 1 weight unit = 1 byte + a small overhead"